vyper/cli/vyper_compile.py

```diff
@@ -143,65 +154,89 @@ def exc_handler(contract_name: ContractName, exception: Exception) -> None:
+        interface_codes = extract_file_interface_imports(code)
         for interface_name, interface_path in interface_codes.items():
-            file_path = Path(interface_path.replace('.', '/')).resolve()
 
-            try:
-                suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists())
-            except StopIteration:
-                raise Exception(
-                    f'Imported interface "{interface_path}{{.vy,.json}}" does not exist.'
+            base_paths = [parent_path]
+            if not interface_path.startswith('.'):
+                base_paths.append(root_path)
+            elif interface_path.startswith('../') and parent_path == root_path:
+                raise FileNotFoundError(
+                    f"{file_path} - Cannot perform relative import outside of base folder"
                 )
 
-            valid_path = file_path.with_suffix(suffix)
+            valid_path = get_interface_file_path(base_paths, interface_path)
             with valid_path.open() as fh:
                 code = fh.read()
                 if valid_path.suffix == '.json':
-                    interfaces[interface_name] = {
+                    interfaces[file_path][interface_name] = {
                         'type': 'json',
                         'code': json.loads(code.encode())
                     }
                 else:
-                    interfaces[interface_name] = {
+                    interfaces[file_path][interface_name] = {
                         'type': 'vyper',
                         'code': code
                     }
 
     return interfaces
 
 
+def get_interface_file_path(base_paths: Sequence, import_path: str) -> Path:
+    relative_path = Path(import_path)
+    for path in base_paths:
+        file_path = path.joinpath(relative_path)
+        try:
+            suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists())
+            return file_path.with_suffix(suffix)
+        except StopIteration:
+            continue
+    raise Exception(
+        f'Imported interface "{import_path}{{.vy,.json}}" does not exist.'
+    )
+
+
 def compile_files(input_files: Iterable[str],
                   output_formats: Sequence[str],
+                  root_folder: str = '.',
                   show_gas_estimates: bool = False) -> OrderedDict:
 
     if show_gas_estimates:
         parser_utils.LLLnode.repr_show_gas = True
 
+    root_path = Path(root_folder).resolve()
+    if not root_path.exists():
+        raise FileNotFoundError(f"Invalid root path - '{root_path.as_posix()}' does not exist")
+
     codes: ContractCodes = OrderedDict()
     for file_name in input_files:
-        with open(file_name) as fh:
-            codes[file_name] = fh.read()
+        file_path = Path(file_name).resolve()
+        file_str = file_path.relative_to(root_path).as_posix()
+        with file_path.open() as fh:
+            codes[file_str] = fh.read()
 
+    version = False
```

Comment:
Rename the this variable to something more descriptive like `set_version`  / `show_version` ?